include/adobe/contract_checks.hpp

```diff
@@ -81,6 +82,25 @@ namespace detail {
 
 #define INTERNAL_ADOBE_CONTRACT_VIOLATED(condition, kind, file, line, message) ::std::terminate()
 
+#elif INTERNAL_ADOBE_CONTRACT_VIOLATION_BEHAVIOR == INTERNAL_ADOBE_CONTRACT_VIOLATION_custom_verbose
+
+#define INTERNAL_ADOBE_CONTRACT_VIOLATED(condition, kind, file, line, message) \
+  ::adobe_contract_violated_verbose(condition, kind, file, line, message)
+
+[[noreturn]] extern void adobe_contract_violated_verbose(const char *condition,
```

Comment:
Documentation says these are in namespace `adobe` and un-prefixed.  You need to make them consistent.

Reply:
Documentation does not say they are in namespace adobe and unprefixed. I'm not sure why the confusion. We could put them in the adobe namespace bug I always thought adding components to library namespace was questionable.

Reply:
I was confused by the declaration in the docs which textually has one use of `adobe_…` and another of `adobe::`; I missed the first one, and was also biased by the way we used to do it.

You aren't adding a component to the library namespace; you're providing the implementation of something that's already in that namespace.  In fact I think adding something to the global namespace could be seen as more questionable.